### DISCUSSION TITLE
Getting to previuos app after a notification - how to handle?

### DISCUSSION THREAD

**Discussion by 0ffTopic** [zepp-health/zeppos-forum#342] [Q&A]
Working on an app for ZeppOS v1.

I'm using this code on my app.js to keep my app open:
```
App({
  globalData: {},
  onCreate(options) {
    console.log('app on create invoke')
    hmApp.setScreenKeep(true);
  },

  onDestroy(options) {
    console.log('app on destroy invoke')
    hmApp.setScreenKeep(false);
  }
})
```

the user interacts with the app until the screen goes off, the user press the button and the watch screen turns on and I have all my previous data.
The problem here is, the user loses all data when going back to the app after a notification.

I'm trying to use a simple file to save all data and load it when the app is on the foreground again, but with no success (following ToDo app template code without the communication).
Is there other way to keep data or to prevent notifications when the app is running?

Thanks

**Comment by silver-zepp** [comment]
I would suggest implementing a `1-5 minute` autosave feature.
And to make it simple, you can use [easy storage](https://github.com/silver-zepp/zeppos-easy-storage/blob/master/easy-storage/v1/easy-storage.js). 
It's a portable one-file solution that needs no dependencies.

```js
import EasyStorage from "easy-storage";
const storage = new EasyStorage();

storage.setKey(...); // autosave the data on interval
...

Page({
    onDestroy(){
        storage.setKey(...); // ... and on app destroy
    }
})
...

storage.getKey(...); // retrieve it on restart
```

**Comment by 0ffTopic** [comment]
Thanks for your suggestion.

I've implemented this but I'm facing errors and it does not behave the expected way.
These are the logs from the simulator:
```
5/22 [0]13:48:31.366 I/quickjs-libc SBEngine: [2024-05-22T21:48:31.365Z] [jsfwk.debug] [tenis-padel-score] [-] read
{"t1Score":0,"t2Score":30,"t1Game":2,"t2Game":1,"isTieBreak":false,"setNumber":0,"set":["2 - 1","0 - 0","0 - 0","0 - 0","0 - 0"]

5/22 [0]13:50:54.641 I/quickjs-libc SBEngine: TypeError: not a function

5/22 [0]13:50:54.641 I/quickjs-libc SBEngine: 

5/22 [0]13:50:54.641 I/js_rt_continer SBEngine: runContinerFunc: run Conti
5/22 [0]13:50:54.643 I/quickjs-libc SBEngine: 

[2024-05-22T21:50:54.642Z] [jsfwk.debug] [tenis-padel-score] [-] page onDestroy invoked 
```

I'm giving you access to the repository, if you can take a quick look it would be great. 
Also is there a way to access logs on the watch?

**Comment by 0ffTopic** [comment]
Still trying to figure out what is not working.

Does EasyStorage works with ZeppOS v1? 
I am able to save a key, but if I close and reopen the app (emulator or device) it does not load any data because the getKey returns nothing.

Do I need any extra config or something? I've looked to the examples and add some dependencies to app.json but still no luck.

**Comment by silver-zepp** [comment]
Yes, `EasyStorage` works on zos v1, but only that particular version that is linked above. 

I believe what you are facing @0ffTopic is the bug on zos v1 that is destroying storage contents on each save. ZeppOS v1 has reached its EOL, so unfortunately you can only use workarounds this time.

Suggestions:
---
1. Make sure you do not trigger project saving/overwriting. Press `Home` on your keyboard while the Emulator is in focus to exit the app. Then press it again -> find the app in the app drawer and test if your save persists
2. Use real hardware only, which doesn't have this bug
3. Use [VisLog](https://github.com/silver-zepp/zeppos-visual-logger) to get the logs on the screen and help you with debugging (make sure to use the v1 version) 
4. Print the contents of your EasyStorage on boot with `vis.log(storage.getContents());`
